### PR TITLE
Ft7 Adjust lagrange multipliers according to dataset size

### LINKED ISSUE #7: Lagrange multipliers need to be manually scaled with the size of the dataset
## Description

Currently we need to use larger multipliers for larger datasets, and smaller multipliers for smaller datasets.

We could potentially just multiply the gradient flowing to each sample from the constraint loss by the number of data points (or simply not divide this loss by the number of data points).

This isn't exactly theoretically sound AFAIK, as the true gradient from FPR or FNR constraints does depend on the number of data points... We could just implement it and see if performance is affected.

### PR DESCRIPTION
**Pull request by garsenio**
Closes #7

### CODE REVIEW

**Review by AndreFCruz — COMMENTED** [review]
I think this PR has to be thoroughly tested.

Namely, the unit and system tests are indeed currently passing but we didn't even update the multiplier_learning_rate for those tests.

Ideally some tests _should fail_ after this change, as the previous values of 50k do not make sense now that we're scaling the gradients according to the dataset size.

It's expected that where we previously used a mult. learning rate of a few thousands, we should now use values between 0 and 1.

We also need to remember to update the default values for the `multiplier_learning_rate` to this new value, I suggest a default of `0.1`

File: include/LightGBM/objective_function.h [hunk 1]
```diff
@@ -566,6 +566,9 @@ class ConstrainedObjectiveFunction : public ObjectiveFunction
         // Update index of multipliers to be used for next constraints
         multipliers_base_index += 1;
       }
+
+      gradients[i] *= num_data_;
```
Comment (AndreFCruz) [inline include/LightGBM/objective_function.h]: This multiplication has to be done in each if clause, for each loss function, instead of at the end.

The `gradients` array is not zeroed out at the beginning of this function, it already has gradients from the predictive loss (and we only want to scale gradients from constraint loss).

**Comment by garsenio** [comment]
Merge awaits final testing by @AndreFCruz

**Comment by AndreFCruz** [comment]
Closed in favor of #37
